macaroon.js

```diff
@@ -220,30 +252,48 @@ const base64url = function(buf) {
   Converts a Uint8Array to a bitArray for use by nacl.
   @param {Uint8Array} arr The array to convert.
 */
-function bytesToBitArray(arr) {
-  return sjcl.codec.base64.toBits(nacl.util.encodeBase64(arr));
-}
+const bytesToBits = function(arr) {
+  // See https://github.com/bitwiseshiftleft/sjcl/issues/344 for why
+  // we cannot just use sjcl.codec.bytes.toBits.
+  return sjcl.codec.base64.toBits(naclutil.encodeBase64(arr));
+};
 
 /**
   Converts a bitArray to a Uint8Array.
   @param {bitArray} arr The array to convert.
 */
-function bitArrayToUint8Array(arr) {
-  return nacl.util.decodeBase64(sjcl.codec.base64.fromBits(arr));
-}
+const bitsToBytes = function(arr) {
+  // See https://github.com/bitwiseshiftleft/sjcl/issues/344 for why
+  // we cannot just use sjcl.codec.bytes.toBits.
+  return naclutil.decodeBase64(sjcl.codec.base64.fromBits(arr));
+};
 
 /**
   Converts a hex to Uint8Array
   @param {String} hex The hex value to convert.
   @return {Uint8Array} The resulting array.
 */
-function hexToBytes(hex) {
+const hexToBytes = function(hex) {
   const arr = new Uint8Array(Math.ceil(hex.length / 2));
   for (let i = 0; i < arr.length; i++) {
     arr[i] = parseInt(hex.substr(i * 2, 2), 16);
   }
   return arr;
-}
+};
+
+/**
+ * Report whether the argument encodes a valid utf-8 string.
+ * @param {Uint8Array} The bytes to check.
+ * @return {bool} True if the bytes are valid utf-8.
+ */
+const isValidUTF8 = function(bytes) {
+  try {
+    bytesToString(bytes);
+  } catch (e) {
+    return false;
```

Comment:
If it's meaningful and useful error `console.error(e)`.
If the error is caught further up the stack obviously don't need to

Reply:
As discussed, I've just added a comment here.